crates/release-automation/src/crate_selection/mod.rs

```diff
@@ -114,123 +114,67 @@ impl<'a> Crate<'a> {
-
-                        // todo: write a test-case for this
-                        if dep.is_optional() && self.workspace.criteria.exclude_optional_deps {
-                            trace!(
-                                "[{}] excluding optional dependency '{}'",
-                                package.name(),
-                                dep_name,
-                            );
+                    let dep_name = dep.package_name().to_string();
 
-                            continue;
-                        }
-
-                        // ignore dev-dependencies that have no version specified, as they will be ignored by `cargo publish` as well
-                        if let CargoDepKind::Development = dep.kind() {
-                            if !dep.specified_req() {
-                                warn!(
-                                    "[{}] excluding unversioned dev-dependency '{}'",
-                                    package.name(),
-                                    dep_name
-                                );
+                    // todo: write a test-case for this
+                    if dep.is_optional() && self.workspace.criteria.exclude_optional_deps {
+                        trace!(
+                            "[{}] excluding optional dependency '{}'",
+                            package.name(),
+                            dep_name,
+                        );
 
-                                continue;
-                            }
-                        }
+                        continue;
+                    }
 
-                        dependencies
-                            .entry(dep.package_name().to_string())
-                            .or_insert_with(|| LinkedHashSet::<Dependency>::default())
-                            .insert(dep.to_owned());
-
-                        if dep.source_id().local_path().is_some() {
-                            let dep_package = ws_members
-                                .get(&dep.package_name().to_string())
-                                .ok_or(anyhow::format_err!(
-                                    "local dependency {} with path {:?} not found in workspace",
-                                    dep.package_name(),
-                                    dep.source_id().to_string()
-                                ))?;
-                            if dep_package.name() == self.package.name() {
+                    // only consider workspace members
+                    if let Some(dep_package) = ws_members.get(&dep.package_name().to_string()) {
+                        // only consider non-star version requirements
+                        if dep.specified_req() && dep.version_req().to_string() != "*" {
+                            // don't add this package to its own dependencies
+                            if dep_package.name() != self.package.name() {
```

Comment:
as discussed this should be

```suggestion
                            if dep_package.name() != package.name() {
```